lalrpop/src/lexer/nfa/mod.rs

```diff
@@ -225,25 +222,29 @@ impl Nfa {
         accept: NfaStateIndex,
         reject: NfaStateIndex,
     ) -> Result<NfaStateIndex, NfaConstructionError> {
-        match *expr.kind() {
+        match expr.kind() {
             HirKind::Empty => Ok(accept),
 
-            HirKind::Literal(ref l) => {
-                match *l {
-                    // [s0] -otherwise-> [accept]
-                    Literal::Unicode(c) => {
-                        let s0 = self.new_state(StateKind::Neither);
-                        self.push_edge(s0, Test::char(c), accept);
-                        self.push_edge(s0, Other, reject);
-                        Ok(s0)
-                    }
-                    //// Bytes are not supported
-                    Literal::Byte(b) => {
-                        let s0 = self.new_state(StateKind::Neither);
-                        self.push_edge(s0, Test::byte(b), accept);
-                        self.push_edge(s0, Other, reject);
-                        Ok(s0)
-                    }
+            HirKind::Literal(Literal(l)) => {
+                /* // [s0] -otherwise-> [accept]
+                Literal::Unicode(c) => {
+                    let s0 = self.new_state(StateKind::Neither);
+                    self.push_edge(s0, Test::char(c), accept);
+                    self.push_edge(s0, Other, reject);
+                    Ok(s0)
+                } */
+                //// Bytes are not supported
+
+                if l.len() == 1 {
+                    let s0 = self.new_state(StateKind::Neither);
+                    self.push_edge(s0, Test::byte(l[0]), accept);
+                    self.push_edge(s0, Other, reject);
+                    Ok(s0)
+                } else {
+                    let s0 = self.new_state(StateKind::Neither);
+                    self.push_edge(s0, Test::byte(*l.last().unwrap()), accept);
+                    self.push_edge(s0, Other, reject);
+                    self.expr(&Hir::literal(&l[..l.len() - 1]), s0, reject)
```

Comment:
```suggestion
                Ok(l.0.iter().rev().fold(accept, |accept, &b| {
                    let s0 = self.new_state(StateKind::Neither);
                    self.push_edge(s0, Test::char(c), accept);
                    self.push_edge(s0, Other, reject);
                    s0
                }))
```

Reply:
Hmm, is there any way to modify this suggestion in github? I think `Test::char(c)` should be `Test::byte(b)` or `Test::byte(*b)`

Reply:
I edited the suggestion 👍